types/index.d.ts

```diff
@@ -1,34 +1,35 @@
-import { Decimal } from 'decimal.js';
+import { Decimal } from 'decimal.js'
 
-declare const math: math.MathJsStatic;
-export as namespace math;
-export = math;
+declare const math: math.MathJsStatic
+export as namespace math
+export = math
 
-type NoLiteralType<T> =
-    T extends number ? number :
-        T extends string ? string :
-            T extends boolean ? boolean :
-                T;
+type NoLiteralType<T> = T extends number
+  ? number
+  : T extends string
+  ? string
+  : T extends boolean
+  ? boolean
+  : T
 
 declare namespace math {
-  type MathNumericType = number | BigNumber | Fraction | Complex;
-  type MathArray = MathNumericType[] | MathNumericType[][];
-  type MathCollection = MathArray | Matrix;
-  type MathType = MathNumericType | Unit | MathCollection;
-  type MathExpression = string | string[] | MathCollection;
+  type MathNumericType = number | BigNumber | Fraction | Complex
+  type MathArray = MathNumericType[] | MathNumericType[][]
+  type MathCollection = MathArray | Matrix
+  type MathType = MathNumericType | Unit | MathCollection
+  type MathExpression = string | string[] | MathCollection
 
-  type FactoryFunction<T> = (scope: any) => T;
+  type FactoryFunction<T> = (scope: any) => T
 
```

Comment:
These warnings are reported in the scripts. The ESLint CLI has a `--max-warnings` option, so we can enforce "no warnings" if we want to, or we could set the limit at 106, which is the current number of warnings (I believe 105 of them are explicit any in this file).

I'm slightly torn about what to do with these explicit anys... `index.d.ts` . Options include:
1. suppress them all with single-line comments ... no fun
2. suppress all the warnings in the file ... In general, "any" should be avoided, so suppressing this for the whole file doesn't seem great.
3. do nothing: live the the warnings.

(3) seems like the best option to me. The warnings are annoying, **but they should be**. The annoyance discourages use of `any`, which is the point of the warning.

Reply:
When running `npm run lint` now, I indeed get a large list of warnings about `any`. I agree that in general we should try to avoid using `any`. These warnings will be only useful when we actually actively solve them right away. But in this case I expect the `any` types to stay there for the foreseeable future. So keeping the warnings (option 3) will result in training yourself to ignore those warnings, since you're not going to act on it, making them effectively useless (and only cause noise and a vague feeling that "there are problems"). So I do not prefer option (3) for that reason.

I do have a slight preference for your option (1), but it is also fine with me to just suppress all `any` warnings in the whole file for now (2). What do you think?

Reply:
> I do have a slight preference for your option (1)

Cool, I'll suppress them individually.

In that case, there will be 0 warnings. Do you think eslint should be run with `--max-warnings 0`?

Reply:
Yes sounds good 👍